### PR TITLE
feat(arch): move && simplify component Message

### PR DESCRIPTION
**Pull request by Leotheluck**
⚠ BREAKING CHANGES ⚠
* OsdsMessage is no longer part of @ovhcloud/ods-stencil package
* OsdsMessage is now accessible by adding either ods-component-message or ods-components dependencies
* @ovhcloud/ods-component-message lib has been replaced by @ovhcloud/ods-component-message
* @ovhcloud/ods-component-message-react lib has been replaced by @ovhcloud/ods-component-message-react
* @ovhcloud/ods-component-message-vue lib has been replaced by @ovhcloud/ods-component-message- vue
* OdsMessageType has been removed from @ovhcloud/ods-core and is replaced by ODS_MESSAGE_TYPE from @ovhcloud/ods-component-message
* OdsMessageTypeList has been removed from @ovhcloud/ods-core and is replaced by ODS_MESSAGE_TYPES from @ovhcloud/ods-component-message
* OdsMessageTypeUnion has been removed from @ovhcloud/ods-core
* OsdsMessage is now displayed as flex by default
* OsdsMessage flex attribute has been removed
* OsdsMessage inline attribute has been added

### CODE REVIEW

File: packages-new/components/message/src/components/osds-message/constants/default-attributes.ts [hunk 1]
```diff
@@ -0,0 +1,15 @@
+import type { OdsMessageAttribute } from '../interfaces/attributes';
+import { OdsThemeColorIntent } from '@ovhcloud/ods-theming';
+
+const DEFAULT_ATTRIBUTE: OdsMessageAttribute = Object.freeze({
+    color: OdsThemeColorIntent.default,
+    inline: false,
+    icon: undefined,
```
Comment (dpellier) [inline packages-new/components/message/src/components/osds-message/constants/default-attributes.ts:7]: alpha order would be nice

File: packages-new/components/message/src/components/osds-message/core/controller.spec.ts [hunk 2]
```diff
@@ -1,31 +1,37 @@
...
-import { OdsMessageType } from './ods-message-type';
+import { OdsClearLoggerSpy, OdsInitializeLoggerSpy, OdsLoggerSpyReferences } from '@ovhcloud/ods-common-testing';
+import { Ods, OdsLogger } from '@ovhcloud/ods-common-core';
+import { OsdsMessage } from '../osds-message';
+import { OdsMessageController } from './controller';
+import { ODS_MESSAGE_TYPE } from '../constants/message-type';
 import { OdsThemeColorIntent } from '@ovhcloud/ods-theming';
-import { odsMessageDefaultAttributes } from './ods-message-default-attributes';
+import { DEFAULT_ATTRIBUTE } from '../constants/default-attributes';
+import type { EventEmitter } from '@stencil/core';
```
Comment (dpellier) [inline packages-new/components/message/src/components/osds-message/core/controller.spec.ts:8]: type on top

File: packages-new/components/message/src/components/osds-message/interfaces/attributes.ts [hunk 3]
```diff
@@ -23,9 +22,13 @@ export interface OdsMessageAttributes extends OdsComponentAttributes {
...
    * The message type
    */
-  type?: OdsMessageType
+  type?: ODS_MESSAGE_TYPE
   /**
    * Message is in contrasterd mode or not
    */
-   contrasted?: boolean
-  }
+  contrasted?: boolean
```
Comment (dpellier) [inline packages-new/components/message/src/components/osds-message/interfaces/attributes.ts:29]: alpha order

File: packages-new/components/message/src/components/osds-message/osds-message.e2e.screenshot.ts [hunk 4]
```diff
@@ -1,49 +1,45 @@
...
 
 describe('e2e:osds-message', () => {
   let page: E2EPage;
   let el: E2EElement;
 
-  async function setup({ attributes = {}, html = `` }: { attributes?: Partial<OdsMessageAttributes>, html?: string } = {}) {
-    const minimalAttributes: OdsMessageAttributes = OdsCreateAttributes(attributes, odsMessageBaseAttributes);
-    const stringAttributes = OdsComponentAttributes2StringAttributes<OdsMessageAttributes>(minimalAttributes, odsMessageDefaultAttributes);
+  async function setup({ attributes = {}, onPage }: { attributes?: Partial<OdsMessageAttribute>, html?: string, onPage?: ({ page }: { page: E2EPage }) => void } = {}) {
+    const stringAttributes = odsComponentAttributes2StringAttributes<OdsMessageAttribute>({ ...attributes }, DEFAULT_ATTRIBUTE);
```
Comment (dpellier) [inline packages-new/components/message/src/components/osds-message/osds-message.e2e.screenshot.ts:15]: destructuring not needed on attributes

File: packages-new/components/message/src/components/osds-message/osds-message.e2e.ts [hunk 5]
```diff
@@ -1,24 +1,24 @@
...
+import { DEFAULT_ATTRIBUTE } from './constants/default-attributes';
 
 describe('e2e:osds-message', () => {
   let page: E2EPage;
   let el: E2EElement;
 
-  async function setup({ attributes = {} }: { attributes?: Partial<OdsMessageAttributes> } = {}) {
-    const minimalAttributes: OdsMessageAttributes = OdsCreateAttributes(attributes, odsMessageBaseAttributes);
-    const stringAttributes = OdsComponentAttributes2StringAttributes<OdsMessageAttributes>(minimalAttributes, odsMessageDefaultAttributes);
+  // async function setup({ attributes = {} }: { attributes?: Partial<OdsMessageAttribute> } = {}) {
```
Comment (dpellier) [inline packages-new/components/message/src/components/osds-message/osds-message.e2e.ts:14]: remove

File: packages-new/components/message/src/components/osds-message/osds-message.e2e.ts [hunk 6]
```diff
@@ -1,24 +1,24 @@
...
+  async function setup({ attributes = {}, onPage }: { attributes?: Partial<OdsMessageAttribute>, html?: string, onPage?: ({ page }: { page: E2EPage }) => void } = {}) {
+    const stringAttributes = odsComponentAttributes2StringAttributes<OdsMessageAttribute>({ ...attributes }, DEFAULT_ATTRIBUTE);
```
Comment (dpellier) [inline packages-new/components/message/src/components/osds-message/osds-message.e2e.ts:16]: no destr

File: packages-new/components/message/src/components/osds-message/osds-message.spec.ts [hunk 7]
```diff
@@ -0,0 +1,142 @@
...
+  let root: HTMLElement | undefined;
+  let instance: OsdsMessage;
+  let controller: OdsMessageController;
+
+  afterEach(() => {
+    jest.clearAllMocks();
+  });
+
+  async function setup({ attributes = {} }: { attributes?: Partial<OdsMessageAttribute> } = {}) {
+    const stringAttributes = odsComponentAttributes2StringAttributes<OdsMessageAttribute>({ ...attributes }, DEFAULT_ATTRIBUTE);
```
Comment (dpellier) [inline packages-new/components/message/src/components/osds-message/osds-message.spec.ts:25]: no destr

File: packages-new/components/message/src/components/osds-message/osds-message.tsx [hunk 8]
```diff
@@ -1,15 +1,13 @@
...
+import type { OdsThemeColorIntent } from '@ovhcloud/ods-theming';
+import type { EventEmitter } from '@stencil/core';
+import type { OdsMessageAttribute } from './interfaces/attributes';
+import type { OdsMessageEvent } from './interfaces/events';
+import { Component, Event, Host, Prop, Watch, h } from '@stencil/core';
+import { DEFAULT_ATTRIBUTE } from './constants/default-attributes';
+import { ODS_MESSAGE_TYPE } from './constants/message-type';
 import { ODS_ICON_NAME, ODS_ICON_SIZE } from '@ovhcloud/ods-component-icon';
+import { OdsMessageController } from './core/controller';
+// import { OdsStencilEvents, OdsStencilMethods } from '@ovhcloud/ods-stencil/libraries/stencil-core';
```
Comment (dpellier) [inline packages-new/components/message/src/components/osds-message/osds-message.tsx:10]: remove

**Comment by ovh-cds** [comment]
CDS Report build#1226.0 ✘  
* 
  * build ✘

**Review by dpellier — APPROVED** [review]

**Pull request merged**